src/hutil.ml

```diff
@@ -18,40 +16,25 @@ value commd_no_params conf =
 
 value link_to_referer conf =
   let referer = Util.get_referer conf in
+  let retour = Util.capitale (Util.transl conf "back") in
   if referer <> "" then
-    let fname = "left.png" in
-    let wid_hei =
-      match Util.image_size (Util.image_file_name fname) with
-      [ Some (wid, hei) ->
-          " width=\"" ^ string_of_int wid ^ "\" height=\"" ^
-          string_of_int hei ^ "\""
-      | None -> "" ]
-    in
-    sprintf "<a href=\"%s\"><img src=\"%s/%s\"%s alt=\"&lt;&lt;\" title=\"&lt;&lt;\"%s></a>\n"
-      referer (Util.image_prefix conf) fname wid_hei conf.xhs
+    sprintf "<a href=\"%s\"><span class=\"fa fa-arrow-left fa-lg\" title=\"%s\"></span></a>\n" referer retour
   else ""
 ;
 
 value gen_print_link_to_welcome f conf right_aligned =
   if conf.cancel_links then ()
   else do {
-    let fname = up_fname conf in
-    let wid_hei =
-      match Util.image_size (Util.image_file_name fname) with
-      [ Some (wid, hei) ->
-          " width=\"" ^ string_of_int wid ^ "\" height=\"" ^
-          string_of_int hei ^ "\""
-      | None -> "" ]
-    in
     if right_aligned then
-      Wserver.printf "<div style=\"float:%s\">\n" conf.right
+      Wserver.printf "<div class=\"btn-group float-%s mt-2\">\n" conf.right
     else Wserver.printf "<p>\n";
     f ();
     let str = link_to_referer conf in
     if str = "" then () else Wserver.printf "%s" str;
     Wserver.printf "<a href=\"%s\">" (commd_no_params conf);
-    Wserver.printf "<img src=\"%s/%s\"%s alt=\"^^\" title=\"^^\"%s>"
-      (Util.image_prefix conf) fname wid_hei conf.xhs;
+    Wserver.printf "<span class=\"fa fa-home fa-lg ml-1 px-0\" title=\"";
+    Wserver.printf "%s" (Util.capitale (Util.transl conf "home"));
+    Wserver.printf "\"></span>";
```

Comment:
wow this is very difficult to read. I did not get it was the same html element you were printing.